### PR TITLE
[MAINTENANCE] Refactor docker test CI steps into jobs.

### PR DESCRIPTION
**Pull request by billdirks**
This PR reorganizes the docker tests from using steps to jobs. This establishing a pattern I can use to integrate the other currently non-docker tests as jobs so they can run simultaneously instead of in series.

In addition, this adds a new step to create the image suffix. This is so our docker images (there is 1 per python version) for a particular pipeline run all have the same suffix so we can more easily group them together.

Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

Changes proposed in this pull request:
-
-
-

After submitting your PR, CI checks will run and @cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -122,8 +122,18 @@ stages:
     pool:
       vmImage: 'ubuntu-latest'
     jobs:
-      - job: build_push_test_
-        timeoutInMinutes: 120
+      - job: make_suffix
```
Comment (billdirks) [inline azure-pipelines.yml:125]: I've added a new `make_suffix` job so we can share this suffix among each docker image (there is 1 docker image per python version).

Comment (Shinnnyshinshin) [reply azure-pipelines.yml:125]: ❤️

File: azure-pipelines.yml [hunk 2]
```diff
@@ -132,43 +142,59 @@ stages:
...
           - bash: |
-              EPOCH=`date +%s`
-              TAG="$(python.version)_develop_${EPOCH}_${RANDOM}"
+              DOCKER_TAG="$(python.version)_$(IMAGE_SUFFIX)"
               # We repull the develop branch. We could build the local branch that is already present but we should
               # prune it's depth since we don't need the git history.
-              CMD="docker buildx build -f docker/Dockerfile.tests --tag greatexpectations/test:${TAG} --target azure --build-arg PYTHON_VERSION=$(python.version) --build-arg SOURCE=github --build-arg BRANCH=develop ."
+              CMD="docker buildx build -f docker/Dockerfile.tests --tag greatexpectations/test:${DOCKER_TAG} "
+              CMD+="--target azure --build-arg PYTHON_VERSION=$(python.version) --build-arg SOURCE=github "
+              CMD+="--build-arg BRANCH=develop ."
```
Comment (Shinnnyshinshin) [inline azure-pipelines.yml:154]: love it

**Review by Shinnnyshinshin — APPROVED** [review]
Thank you @billdirks. Looks good on my end

**Pull request merged**